src/hooks/use-field-selection.tsx

```diff
@@ -0,0 +1,45 @@
+import React, { createContext, useContext, useMemo, ReactNode } from 'react';
+
+import { FieldId } from '@/types';
+
+interface FieldSelectionContextType {
+  enabled: boolean;
+  fieldProps:
+    | {
+        onClick: (event: React.MouseEvent, params: { id: FieldId; nodeId: string }) => void;
```

Comment:
Nit: Can we extract this out to its own type, it's being used further down the file also